src/builder.rs

```diff
@@ -645,43 +754,34 @@ impl<'ctx> Builder<'ctx> {
 
         let mut index_values: Vec<LLVMValueRef> = ordered_indexes.iter().map(|val| val.as_value_ref()).collect();
 
-        // This ugly cfg specification is due to limitation of custom attributes (for more information, see https://github.com/rust-lang/rust/issues/54727).
-        // Once custom attriutes inside methods are enabled, this should be replaced with #[llvm_version(14.0..=latest)]
-        #[cfg(not(any(
-            feature = "llvm4-0",
-            feature = "llvm5-0",
-            feature = "llvm6-0",
-            feature = "llvm7-0",
-            feature = "llvm8-0",
-            feature = "llvm9-0",
-            feature = "llvm10-0",
-            feature = "llvm11-0",
-            feature = "llvm12-0",
-            feature = "llvm13-0",
-        )))]
-        let value = LLVMBuildGEP2(
+        let value = LLVMBuildGEP(
             self.builder,
-            ptr.get_type().get_element_type().as_type_ref(),
             ptr.as_value_ref(),
             index_values.as_mut_ptr(),
             index_values.len() as u32,
             c_string.as_ptr(),
         );
 
-        #[cfg(any(
-            feature = "llvm4-0",
-            feature = "llvm5-0",
-            feature = "llvm6-0",
-            feature = "llvm7-0",
-            feature = "llvm8-0",
-            feature = "llvm9-0",
-            feature = "llvm10-0",
-            feature = "llvm11-0",
-            feature = "llvm12-0",
-            feature = "llvm13-0",
-        ))]
-        let value = LLVMBuildGEP(
+        PointerValue::new(value)
+    }
+
+    // REVIEW: Doesn't GEP work on array too?
+    /// GEP is very likely to segfault if indexes are used incorrectly, and is therefore an unsafe function. Maybe we can change this in the future.
+    #[llvm_versions(14.0..=latest)]
+    pub unsafe fn build_gep<T: BasicType<'ctx>>(
+        &self,
+        ty: T,
```

Comment:
can we use a more explicit name here? I think the type here should be "the type that the pointer points at"

Reply:
Makes sense.
Feel free to rename to `pointee_ty`.